### ISSUE TITLE
NativeAOT targets use `PlatformTarget` to find appropriate runtime pack, which is often wrong

### ISSUE THREAD

**Issue by agocke** [dotnet/runtimelab#1013]
The problem is that PlatformTarget is what's normally passed to the C# compiler to choose the appropriate output architecture for the PE file. This is often not the same as the target architecture passed by RuntimeIdentifier because many projects choose to publish for a particular platform but use IL-only binaries, so `PlatformTarget=AnyCPU`.

This seems to result in the ILCompiler package being unable to find the runtime pack and producing errors like:

```
/home/andy/.nuget/packages/microsoft.dotnet.ilcompiler/6.0.0-preview.5.21223.3/build/Microsoft.NETCore.Native.Publish.targets(74,5): error : The PrivateSdkAssemblies ItemGroup is required for _ComputeAssembliesToCompileToNative [/home/andy/code/runtime/src/coreclr/tools/aot/crossgen2/crossgen2.csproj]
```

I think the right long-term fix for this would be to try to share the code that ReadyToRun uses to find their runtime pack.

**Comment by AntonLapounov** [comment]
Here is how `PlatformTarget` is inferred from `RuntimeIdentifier` in the SDK: https://github.com/dotnet/sdk/blob/0df16f2d9c179032be57ac213c4248e5621d1846/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets#L82-L86.  We can do the same in our `.targets` files.

**Comment by AntonLapounov** [comment]
Fixed by #1072.  Regarding the long-term fix, feel free to open a new issue with more details.
